components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt

```diff
@@ -345,10 +346,29 @@ class GeckoEngine(
                 extension -> GeckoWebExtension(extension, runtime)
             } ?: emptyList()
 
+            val webExtPrivateBrowsingKeyMigrated =
```

Comment:
nit: ++ :) Using our 120 line limit this would be easier to read:
```kotlin
val privateBrowsingMigrated = sharedPref.getBoolean(MOZAC_HAS_RESET_WEBEXT_PRIVATE_BROWSING_DEFAULT, false)
extensions.forEach { extension ->
    // As a workaround for https://bugzilla.mozilla.org/show_bug.cgi?id=1621385,
    // we set all installed extensions to be allowed in private browsing mode.
    // We need to revert back to false which is now the default.
    if (!extension.isBuiltIn() && extension.isAllowedInPrivateBrowsing() && !privateBrowsingMigrated) {
      setAllowedInPrivateBrowsing(extension, false)
    }

    extension.registerActionHandler(webExtensionActionHandler)
    extension.registerTabHandler(webExtensionTabHandler)
}

if (!privateBrowsingMigrated) {
  sharedPref.edit().putBoolean(MOZAC_HAS_RESET_WEBEXT_PRIVATE_BROWSING_DEFAULT, true).apply()
}
```